src/main/java/com/blazemeter/jmeter/http2/sampler/HTTP2Request.java

```diff
@@ -187,6 +197,11 @@ private void errorResult(Throwable e, HTTP2SampleResult res) {
     protected void sample(URL url, String method, HTTP2Connection http2Connection,
                           HTTP2SampleResult sampleResult) {
 
+        String urlStr = url.toString();

```

Comment:
Knowing that this variable will be used just once, would be cleaner to just inline it.